src/IECoreScene/MeshAlgoDistributePoints.cpp

```diff
@@ -134,39 +139,93 @@ T triangleInterpolatedPrimVarValue(
 void processInputs(
 	const MeshPrimitive *mesh,
 	const std::string &position, const std::string &uvSet, const std::string &densityMask,
+	const StringAlgo::MatchPattern &primitiveVariables,
 	const Canceller *canceller,
-	MeshPrimitivePtr &triangulatedMesh,
-	PrimitiveVariable &positionVar, PrimitiveVariable &uvVar,
-	PrimitiveVariable &faceAreaVar, PrimitiveVariable &textureAreaVar, PrimitiveVariable &densityVar
+	MeshPrimitivePtr &processedMesh,
+	PrimitiveVariable &uvVar, PrimitiveVariable &densityVar,
+	PrimitiveVariable &faceAreaVar, PrimitiveVariable &textureAreaVar
 )
 {
 	if( !mesh )
 	{
 		throw InvalidArgumentException( "MeshAlgo::distributePoints : The input mesh is not valid" );
 	}
 
-	// \todo - remove unused primvars
-	triangulatedMesh = MeshAlgo::triangulate( mesh, canceller );
+	IECoreScene::MeshPrimitivePtr meshWithUsedPrimVars = new MeshPrimitive();
+
+	// We need the topolgy of the source mesh to triangulate it
+	meshWithUsedPrimVars->setTopologyUnchecked( mesh->verticesPerFace(), mesh->vertexIds(), mesh->variableSize( PrimitiveVariable::Vertex ), mesh->interpolation() );
+
+	// Note that we do not transfer creases or corners - they do not affect distribution of points. If we were
+	// to add support for distributing onto the limit surface of a subdiv, then we might need to keep those ...
+	// but that would need to happen on an untriangulated mesh anyway.
+
+	// Transfer the subset of variables that we need
+	for( auto var : mesh->variables )
+	{
+		if( var.first == position )
+		{
+			// \todo - is it a problem that this P that we are setting up solely to be passed in
+			// triangulate ( where it isn't used ), may not have a Point interpretation set correctly?
```

Comment:
> (where it isn't used)

I don't understand this comment - it _is_ used in `triangulate()`, otherwise we wouldn't be passing it would we?

Reply:
I'm not really following all this juggling of `P` and `position` - it seems _painful_. Wasn't it broken before anyway? Since we don't use it in Gaffer, I'd be perfectly happy to ignore it if it simplifies the code. Maybe we can just throw if `position != "P"` and then remove the parameter in 10.5?

Reply:
Yeah, it is fairly painful, and that's after going through a couple of rearrangements that felt even worse. I'm happy to remove this feature if it feels unnecessary. It feels like a bit of a shame to not remove the parameter already in the new signature, since we already need to add a new signature and deprecate the old one - but unfortunately since MatchPattern and string are actually the same type, and the place for "primitiveVariables" in the arguments really does seem to be the same as where "position" used to be, I guess there really isn't a good way to differentiate them.  Throwing is probably good enough for now ... I guess we don't really much about non-Gaffer uses anyway.

Reply:
We could perhaps have called the new function `scatterPoints()` instead. But actually I think we have completely misinterpreted the intention of the `position` argument, and our "fix" for it was wrong.

If you go back to the [origins](https://github.com/ImageEngine/cortex/commit/eaa8243269ef23262f3f20ccf84db337f9a44cc2#diff-a27f3120a622be0b9a2f44250a807354c8c3c60e0e930126ca8adfc97235c764) of this code, you'll see that it was ported from an op, and that in the op the parameter was called `pRefPrimVarName`. That explains why `position` was only being used to calculate the face areas, and _not_ being used to feed the output points. It's only purpose is to get a stable distribution on a deforming mesh - like you'd need for fur follicles on a creature, for instance. So we definitely need to keep the parameter (although we should probably rename it), and we definitely need to keep its original meaning. Hopefully that also means some of the juggling can be removed?

Reply:
Oh, yeah, I was pretty unsure about changing the meaning of it when I asked you, but I guess we were both mislead by the name + lack of tests. And yeah, it should be more straightforward to make it work that way. Does "refPosition" sound good? Or "referencePosition"?

Reply:
Yeah, either of those sounds good - a test case to guide our future selves would be handy too.